src/main.py

```diff
@@ -0,0 +1,39 @@
+from dotenv import load_dotenv
+from importlib import reload
+from supabase import create_client, Client
+from tkinter import messagebox
+import os
+
+# Run splash screen and player registration GUI
+import splash
+import register
+
+# Load environment variables
+load_dotenv()
+
+# Create a new Supabase client
+url: str = os.environ.get("SUPABASE_URL")
+key: str = os.environ.get("SUPABASE_KEY")
+supabase: Client = create_client(url, key)
+
+# Get user input from GUI
+info = register.info
+
+# Check whether user already exists in user table
+user = supabase.table("players").select("*").eq("username", info["username"]).execute()
+
+# While user already exists, re-run registration GUI
+while (user.data != []):
+    # Display error message
+    messagebox.showerror("Error", "Username already exists")
+
+    # Re-run registration GUI
+    reload(register)
+    info = register.info
+    
+    # Re-run query to check whether user already exists
+    user = supabase.table("players").select("*").eq("username", info["username"]).execute()
+
+# Insert player into table and display success message
+supabase.table("players").insert([info]).execute()
+messagebox.showinfo("Success", "User successfully registered!")
```

Comment:
Throwing segmentation fault on MacOS only.

Reply:
<img width="1115" alt="Screenshot 2023-08-28 at 12 16 16 PM" src="https://github.com/cadedupont/photon/assets/98860495/83254a74-900c-468b-831d-ea8d6c61eec1">
Seems to make the POST request successfully and user is added to database, but neither error nor success message boxes are displayed on screen (only on Windows).

Reply:
Since the issue is platform dependent it would seem like the system set limit for memory allocation may be an issue as referenced here https://stackoverflow.com/questions/10035541/what-causes-a-python-segmentation-fault. But I find it odd that it is only faulting after the post request is made as the large amount of data would most likely be in the get request, unless there is some lazy evaluation going on here.

Reply:
Read up that this package addresses some known MacOS tkinter issues: https://pypi.org/project/tkmacosx/. We could attempt to download and import packages based on OS or operate under the assumption that the software is only available on Windows.

Reply:
@cadedupont  Is it worth looking at a different UI library all together then possibly? Just curious overall.

Reply:
An alternative could be something like PyQt5:
https://pypi.org/project/PyQt5/#:~:text=PyQt5%20is%20a%20comprehensive%20set,including%20iOS%20and%20Android.
https://wiki.python.org/moin/PyQt
Might be worth a shot. Willing to block this PR and attempt to port over on another branch.

Reply:
Might just try a quick test and see if the issue persists. If it does we know something else is broken and if not we will know that is a possible route to go down.